client/helpers/parts.ts

```diff
@@ -24,6 +25,38 @@ export const createPartFromFile = async (
   await sleep(1000);
 
   const req = args.createPartReq(fileId);
+  const suppliedId = req.data.attributes.suppliedId;
+  const suppliedRevisionId = req.data.attributes.suppliedRevisionId;
+
+  // TODO: Update once we can filter by both part and part-revision suppliedIds with one API call
+  const getPartRes = await args.client.parts.getParts(undefined, 1, [
+    suppliedId,
+  ]);
+  throwOnError(getPartRes, `Error getting parts by suppliedId '${suppliedId}'`);
```

Comment:
Is it possible to wrap the Axios instance that is used by the client to provide this behavior? Seems like this pattern of throwing on non-success codes will be difficult to enforce moving forward.

Reply:
We can, but that would catch the errors for anyone using the client and I'm not sure everyone would want to handle the errors in the same way.

Reply:
Maybe food for thought then, but not throwing an error is almost never what I want. I understand why these low-level request libraries don't throw errors. As a consumer though, this default approach basically forces me to check the response code for every API call I make, then switch on the result in order to parse the success or error data.

IMO, `Promise.then().catch()` is a better pattern for this. Could also give the consumer an option on what style they want.

Reply:
I added a note for myself to give this some thought and can follow up with you.